components/modal/README.md

```diff
@@ -0,0 +1,176 @@
+Modal
+=======
+
+The modal is used to create an accessible modal over an application.
+
+**Note:** The API for this modal has been mimicked to resemble [`react-modal`](https://github.com/reactjs/react-modal).
+
+## Usage
+
+Render a screen overlay with a modal on top.
+```jsx
+	<Modal
+		title="My Modal"
+		onRequestClose={ closeFunction }
+		isOpen={ openState }
+		aria={ {
+		    describedby: "modal-description",
+		} }
+		>
+		<ModalContent>
+			<p id="modal-description">This modal is meant to be awesome!</p>
+		</ModalConent>
+	</Modal>
+```
+
+## Implement close logic
+
+For the modal to properly work it's important you implement the close logic for the modal properly. The following example shows you how to properly implement a modal.
+
+```js
+const { Component, Fragment } = wp.element;
+const { Modal } = wp.components;
+
+class MyModalWrapper extends Component {
+	constructor() {
+		super( ...arguments );
+		this.state = {
+			isOpen: true,
+		}
+
+		this.openModal = this.openModal.bind( this );
+		this.closeModal = this.closeModal.bind( this );
+	}
+	
+	openModal() {
+		if ( ! this.state.isOpen ) {
+			this.setState( { isOpen: true } );
+		}
+	}
+
+	closeModal() {
+		if ( this.state.isOpen ) {
+			this.setState( { isOpen: false } );
+		}
+	}
+
+	render() {
+		return (
+			<Fragment>
+				<button onClick={ this.openModal }>Open Modal</button>
+				<Modal
+					title="This is my modal"
+					onRequestClose={ this.closeModal }
+					isOpen={ this.state.isOpen }>
```

Comment:
This is something we used to have with Popover, and then was later removed in #5015 because there's a performance impact in allowing the element and its children to be created wastefully (with no intention of ever rendering except by basis of boolean value). Further, there's a point toward consistency between usage of the two components.

Would it be possible to remove this boolean and have the modal render simply by its presence?

Reply:
Also, this `isOpen` prop is not documented.

Reply:
Yes, it is possible to omit the `open` prop. It was just an attempt to keep true to the `react-modal` API, but I think we should not keep trying to mimic it.

Reply:
Yes, I was about to propose the same given that this component is useless when `isOpen` flag is set to `false`.